rules/android/lint/lint_inspect.bzl

```diff
@@ -0,0 +1,79 @@
+load("@grab_bazel_common//rules/android/lint:providers.bzl", "AndroidLintInfo")
+load("@grab_bazel_common//rules/android:utils.bzl", "utils")
+
+def _lint_inspector(ctx):
+    lint_target = ctx.attr.lint_target
+    target = ctx.attr.target
+
+    lint_checks = ctx.attr.lint_target[AndroidLintInfo].info.lint_checks
+    aar_dirs = ctx.attr.lint_target[AndroidLintInfo].info.aars_dir
+
+    executable = ctx.actions.declare_file("lint/%s_inspect.sh" % target.label.name)
+    lint_rules = ctx.actions.declare_file("lint/%s_lint_rules.json" % target.label.name)
+
+    ctx.actions.write(
+        output = executable,
+        content = """
+                    #!/bin/bash
+#                    cat {lint_rules}
+                    echo "\nlint_rules: $(realpath {lint_rules})"
+                            """.format(
+            lint_rules = lint_rules.short_path,
+        ),
+    )
+
+    mnemonic = "AndroidLintInspector"
+    args = ctx.actions.args()
+
+    args.add_joined(
+        "--aar-dirs",
+        aar_dirs,
+        join_with = ",",
+        map_each = utils.to_path,
+    )
+    args.add_joined(
+        "--lint-checks",
+        lint_checks,
+        join_with = ",",
+        map_each = utils.to_path,
+    )
+    args.add("--out-put", lint_rules.path)
+
+    ctx.actions.run(
+        mnemonic = mnemonic,
+        inputs = depset(
+            lint_checks.to_list() + aar_dirs.to_list(),
```

Comment:
We can avoid a `to_list` call here by writing  `transitive = [lint_checks, aar_dirs]`